yt_dlp/extractor/youtube.py

```diff
@@ -4930,26 +4930,30 @@ def get_mobj(url):
-            tab_name = selected_tab.get('title', '')
+            selected_tab_name = selected_tab.get('title', '').lower()
+            requested_tab_name = mobj['tab'][1:]
             if 'no-youtube-channel-redirect' not in compat_opts:
                 if mobj['tab'] == '/live':
                     # Live tab should have redirected to the video
                     raise ExtractorError('The channel is not currently live', expected=True)
-                if mobj['tab'] == '/videos' and tab_name.lower() != mobj['tab'][1:]:
-                    redirect_warning = f'The URL does not have a {mobj["tab"][1:]} tab'
-                    if not mobj['not_channel'] and item_id[:2] == 'UC':
-                        # Topic channels don't have /videos. Use the equivalent playlist instead
-                        pl_id = f'UU{item_id[2:]}'
-                        pl_url = f'https://www.youtube.com/playlist?list={pl_id}'
-                        try:
-                            data, ytcfg = self._extract_data(pl_url, pl_id, ytcfg=ytcfg, fatal=True)
-                        except ExtractorError:
-                            redirect_warning += ' and the playlist redirect gave error'
-                        else:
-                            item_id, url, tab_name = pl_id, pl_url, mobj['tab'][1:]
-                            redirect_warning += f'. Redirecting to playlist {pl_id} instead'
-                    if tab_name.lower() != mobj['tab'][1:]:
-                        redirect_warning += f'. {tab_name} tab is being downloaded instead'
+                if selected_tab_name != requested_tab_name and requested_tab_name != 'featured':
+                    redirect_warning = f'The URL does not have a {requested_tab_name} tab'
+                    if requested_tab_name == 'videos':
+                        if not mobj['not_channel'] and item_id[:2] == 'UC':
+                            # Topic channels don't have /videos. Use the equivalent playlist instead
+                            pl_id = f'UU{item_id[2:]}'
+                            pl_url = f'https://www.youtube.com/playlist?list={pl_id}'
+                            try:
+                                data, ytcfg = self._extract_data(pl_url, pl_id, ytcfg=ytcfg, fatal=True)
+                            except ExtractorError:
+                                redirect_warning += ' and the playlist redirect gave error'
+                            else:
+                                item_id, url, selected_tab_name = pl_id, pl_url, requested_tab_name
+                                redirect_warning += f'. Redirecting to playlist {pl_id} instead'
+                        if selected_tab_name != requested_tab_name:
+                            redirect_warning += f'. {selected_tab_name} tab is being downloaded instead'
+                    elif not mobj['not_channel']:
+                        raise ExtractorError(f'The channel did not present a {requested_tab_name} tab', expected=True)
```

Comment:
```suggestion
                    elif not mobj['not_channel']:
                        raise ExtractorError(redirect_warning, expected=True)
```
Not sure if we need the channel check

Reply:
I wasn't sure about the purpose of the existing channel check, so I added it as a precaution. I didn't think checking tabs made sense on non-channels. If it's redundant then it should probably be removed. But if we have tabs on a non-channel for some reason, then the error might be confusing.

The warning text should probably be reused.

Reply:
Channel check is necessary to avoid conflicts with special URLs like /feed/trending. We might be able to get away with checking for whether a selected tab exists, but the current method makes more sense.